owl/SOMA-OBJ.owl

```diff
@@ -787,6 +812,36 @@
     
 
 
+    <!-- http://www.ease-crc.org/ont/SOMA-OBJ.owl#hasReferencePose -->
+
+    <owl:DatatypeProperty rdf:about="http://www.ease-crc.org/ont/SOMA-OBJ.owl#hasReferencePose">
+        <rdfs:subPropertyOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#hasShapeParameter"/>
+        <rdfs:domain rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#ShapeRegion"/>
+        <rdfs:range rdf:resource="http://www.w3.org/2001/XMLSchema#string"/>
+        <rdfs:comment>Gives the default pose of a shape in its own coordinate system.
+
+It is sometimes the case that a shape described via a mesh or a primitive needs to be translated and/or rotated in order to correspond to the object it is a shape of. This can happen, e.g., when a robot description file places the center point of a link at a different location to the centroid of the mesh giving that link&apos;s shape.</rdfs:comment>
+        <rdfs:label>has reference pose</rdfs:label>
+    </owl:DatatypeProperty>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA-OBJ.owl#hasShapeScale -->
+
+    <owl:DatatypeProperty rdf:about="http://www.ease-crc.org/ont/SOMA-OBJ.owl#hasShapeScale">
+        <rdfs:subPropertyOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#hasShapeParameter"/>
+        <rdfs:domain rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#ShapeRegion"/>
+        <rdfs:range rdf:resource="http://www.w3.org/2001/XMLSchema#string"/>
```

Comment:
Could we use three individual properties with xsd:double range instead (or in addition)?

We could very well also expand the representation of pose data in my opinion. The reason being that the data is not actually stored in the triple store (anymore), but rather in a separate DB collection for TF data, or in URDF files. So the model defined here would mainly be used by KnowRob to access this externally stored data.

Reply:
Yes you are right.

It started with early KnowRob representing 4x4 matrices in the triple store with 16 different properties, one for each slot in the matrix.
That I still think is very excessive and not very meaningful.
I first wanted to avoid these unneeded triples that would not appear in any axiom anyways.
Then I tried to introduce these vector datatypes which are still in SOMA.owl, I think.
But that was never really clean, I would say.

The situation has changed a bit since KnowRob stopped to actually store this information in the triple DB, but rather uses external sources to draw e.g. pose data from TF or shape data from URDF.
So the argument of doing it like that for performance reasons is not entirely valid anymore.

I agree that we should try to be as consistent as possible in how we handle data.
for example, box attributes have at the moment their own property each (i.e. for width, height etc.).
For color and scale we use strings that encode a vector instead, and for poses we have a string attribute that encodes the whole pose.

Surely one disadvantage of strings is that the model does not enforce format, and that applications would need to have their own parser for these strings (in case of not using KnowRob to read the data).